{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "77338551-220e-4a70-bd88-2956446659d2",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@\n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@\n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@\n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@\n",
      "\u001b[?1l\u001b>\u001b[32mImported ./seqtabnochim_Amph16S.fna as DNASequencesDirectoryFormat to ./rep-seqs_Amph16S.qza\u001b[0m\n",
      "\u001b[0m\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@\n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@\n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@\n",
      "\u001b[?1l\u001b>\u001b[32mImported ./table_Amph16S.biom as BIOMV210DirFmt to ./table_Amph16S.qza\u001b[0m\n",
      "\u001b[0m\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@\n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@\n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@\n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@\n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@\n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@\n",
      "\u001b[?1l\u001b>\u001b[32mSaved FeatureTable[Frequency] to: ./table-97_Amph16S.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureData[Sequence] to: ./rep-seqs-97_Amph16S.qza\u001b[0m\n",
      "\u001b[0m\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "# Make consensus sequences (OTU) from ASVs\n",
    "source activate qiime2-amplicon-2024.5\n",
    "\n",
    "############## Amph16S\n",
    "# importation\n",
    "cd /Users/tonyrobinet/sync/BioLav/M2_Ivan/Stage_M2/bioinfo\n",
    "\n",
    "qiime tools import \\\n",
    "--input-path ./seqtabnochim_Amph16S.fna \\\n",
    "--type 'FeatureData[Sequence]' \\\n",
    "--output-path ./rep-seqs_Amph16S.qza\n",
    "\n",
    "biom convert -i ./seqtabnochim_Amph16S.txt -o ./table_Amph16S.biom --table-type=\"OTU table\" --to-hdf5\n",
    "\n",
    "qiime tools import \\\n",
    "--input-path ./table_Amph16S.biom \\\n",
    "--type 'FeatureTable[Frequency]' \\\n",
    "--output-path ./table_Amph16S.qza\n",
    "\n",
    "\n",
    "\n",
    "# vsearch 97% similarity\n",
    "\n",
    "qiime vsearch cluster-features-de-novo \\\n",
    "  --i-table ./table_Amph16S.qza \\\n",
    "  --i-sequences ./rep-seqs_Amph16S.qza \\\n",
    "  --p-perc-identity 0.97 \\\n",
    "  --p-threads 6 \\\n",
    "  --o-clustered-table ./table-97_Amph16S.qza \\\n",
    "  --o-clustered-sequences ./rep-seqs-97_Amph16S.qza"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "023bba05-434c-43e3-b9a5-1da0140538d7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@\n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@\n",
      "\u001b[?1l\u001b>\u001b[32mExported ./table-97_Amph16S.qza as BIOMV210DirFmt to directory ./\u001b[0m\n",
      "\u001b[0m\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@\n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@\n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "# Export OTU tables\n",
    "\n",
    "cd /Users/tonyrobinet/sync/BioLav/M2_Ivan/Stage_M2/bioinfo\n",
    "\n",
    "##############\n",
    "# Amph16S\n",
    "\n",
    "qiime tools export \\\n",
    "    --input-path ./table-97_Amph16S.qza \\\n",
    "    --output-path ./\n",
    "    \n",
    "biom convert \\\n",
    "    -i ./feature-table.biom \\\n",
    "    -o ./feature-table-97_Amph16S.tsv --to-tsv\n",
    "\n",
    "# open the .tsv file with LibreOffice or Excel, select abundances, then Edition > Find and replace,\n",
    "# Seek for ^[0-9] and replace it by & (tick \"regular expressions\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "b1deec7b-3baf-4abc-b63f-7939c550278b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "cd /Users/tonyrobinet/sync/BioLav/M2_Ivan/Stage_M2/bioinfo/RefBase_Amph16S\n",
    "cat ena_*.fasta > ENA_Amphibiens_BZH_16Smito.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "89a53abf-970f-4deb-8250-f855c90af07b",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@\n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@\n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@\n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@\n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@\n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@\n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@\n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@\n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@\n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@\n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@\n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@\n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@\n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@\n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@\n",
      "\u001b[?1l\u001b>662\n",
      "\u001b[?1h\u001b=@\n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@\n",
      "\u001b[?1l\u001b>[INFO]\u001b[0m 1 primer pair loaded\n",
      "\u001b[?1h\u001b=@\n",
      "\u001b[?1l\u001b>573\n",
      "\u001b[?1h\u001b=@\n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@\n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@\n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "cd /Users/tonyrobinet/sync/BioLav/M2_Ivan/Stage_M2/bioinfo/RefBase_Amph16S\n",
    "\n",
    "# NCBI_mitochondrion_AND_amphibia.fasta\n",
    "# RefSeqs_NCBI_Mitochondrion_Amphibia.fasta\n",
    "\n",
    "\n",
    "# sur uniquement les amphibiens\n",
    "# avec \n",
    "\n",
    "#grep \">\" NCBI_mitochondrion_AND_amphibia.fasta | wc -l \n",
    "#seqkit amplicon NCBI_mitochondrion_AND_amphibia.fasta \\\n",
    "#    --threads 8 -F ACGAGAAGACCC..TGGA.CTT -R ATCCAACATCGAGGTCGTAA -r 1:-1 --max-mismatch 4 > NCBI_Amph16S_1070F_1340R.fasta\n",
    "#grep \">\" NCBI_Amph16S_1070F_1340R.fasta | wc -l\n",
    "\n",
    "\n",
    "grep \">\" ENA_Amphibiens_BZH_16Smito.fasta | wc -l #\n",
    "seqkit amplicon ENA_Amphibiens_BZH_16Smito.fasta \\\n",
    "    --threads 8 -F ACGAGAAGACCC..TGGA.CTT -R ATCCAACATCGAGGTCGTAA -r 1:-1 --max-mismatch 4 > ENA_Amphibiens_BZH_Amph16S_1070F_1340R.fasta\n",
    "grep \">\" ENA_Amphibiens_BZH_Amph16S_1070F_1340R.fasta | wc -l # \n",
    "\n",
    "# -F ACGAGAAGACCCYRTGGARCTT -R ATCCAACATCGAGGTCGTAA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "e7ad9fdf-affc-44f9-b6f7-3233773fa59d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>    1970\n",
      "\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@\n",
      "\u001b[?1l\u001b>[INFO]\u001b[0m 1 primer pair loaded\n",
      "\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>     700\n",
      "\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "cd /Users/tonyrobinet/sync/BioLav/M2_Ivan/Stage_M2/bioinfo/RefBase_Amph16S\n",
    "\n",
    "grep \">\" ENA_Amphibiens_invasifs_CROAA_16S.fasta | wc -l #\n",
    "seqkit amplicon ENA_Amphibiens_invasifs_CROAA_16S.fasta \\\n",
    "    --threads 8 -F ACGAGAAGACCC..TGGA.CTT -R ATCCAACATCGAGGTCGTAA -r 1:-1 --max-mismatch 4 > ENA_Amphibiens_invasifs_CROAA_1070F_1340R.fasta\n",
    "grep \">\" ENA_Amphibiens_invasifs_CROAA_1070F_1340R.fasta | wc -l # "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "7f6dc89e-7665-4f4f-ab65-b97d96b990ec",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "# Remove duplicates from ID list\n",
    "cd /Users/tonyrobinet/sync/BioLav/M2_Ivan/Stage_M2/bioinfo/RefBase_Amph16S\n",
    "\n",
    "#a=\"IDs_ENA_Amphibiens_invasifs_CROAA_1070F_1340R.txt\"\n",
    "#cat \"$a\" | awk '{for (i=1;i<=NF;i++) if (!a[$i]++) printf(\"%s%s\",$i,FS)}{printf(\"\\n\")}' > IDs_uniq_ENA_Amphibiens_invasifs_CROAA_1070F_1340R.txt\n",
    "\n",
    "# puis ré-écrire id en 1ere ligne\n",
    "\n",
    "\n",
    "a=\"IDs_NCBI_Mito_Chordata_1070F_1340R.txt\"\n",
    "cat \"$a\" | awk '{for (i=1;i<=NF;i++) if (!a[$i]++) printf(\"%s%s\",$i,FS)}{printf(\"\\n\")}' > IDs_uniq_NCBI_Mito_Chordata_1070F_1340R.txt\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "9d961626-60d9-4a14-80a7-22f05ef79c26",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "## on extrait les identifiants des séquences de la base ad-hoc\n",
    "# NCBI_Amph16S_1070F_1340R.fasta\n",
    "# RefSeqs_NCBI_Amph16S_1070F_1340R.fasta\n",
    "cd /Users/tonyrobinet/sync/BioLav/M2_Ivan/Stage_M2/bioinfo/RefBase_Amph16S\n",
    "\n",
    "#grep \"^>\" ENA_Amphibiens_BZH_Amph16S_1070F_1340R.fasta  | awk -F\"|\" '/\\|/ {print $2}' | sed -e \"s/>//\" > IDs_ENA_Amphibiens_BZH_Amph16S_1070F_1340R.txt\n",
    "grep \"^>\" ENA_Amphibiens_invasifs_CROAA_1070F_1340R.fasta  | awk -F\"|\" '/\\|/ {print $2}' | sed -e \"s/>//\" > IDs_ENA_Amphibiens_invasifs_CROAA_1070F_1340R.txt\n",
    "\n",
    "# il faut ensuite ajouter \"id\" en première ligne (entête de colonne)\n",
    "\n",
    "#grep \"^>\" NCBI_Mito_Chordata_1070F_1340R.fasta  | awk -F\" \" '/\\>/ {print $1}' | sed -e \"s/>//\" > IDs_NCBI_Mito_Chordata_1070F_1340R.txt\n",
    "#grep \"^>\" NCBI_Amph16S_1070F_1340R.fasta  | awk -F\" \" '/\\>/ {print $1}' | sed -e \"s/>//\" > IDs_NCBI_Amph16S_1070F_1340R.txt\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "53f59286-df08-4b0c-b006-961b785694e8",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@\n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@\n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@\n",
      "\u001b[?1l\u001b>\u001b[31m\u001b[1mPlugin error from rescript:\n",
      "\n",
      "  Maximum retries (10) exceeded for HTTP request. Persistent trouble downloading from NCBI. Last exception was\n",
      "  RuntimeError: bad record\n",
      "\n",
      "Debug info has been saved to /var/folders/vz/q1rwtg0113j89y5zgm94slnr0000gn/T/qiime2-q2cli-err-9hvr9oza.log\u001b[0m\n",
      "\u001b[0m\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "## construction du fichier taxo\n",
    "\n",
    "cd /Users/tonyrobinet/sync/BioLav/M2_Ivan/Stage_M2/bioinfo\n",
    "#source activate qiime2-amplicon-2024.5\n",
    "\n",
    "# il faut au préalable ajouter id en 1ere ligne du fichier des identifiants\n",
    "\n",
    "#qiime rescript get-ncbi-data \\\n",
    "#    --p-query \"mitochondrion[All Fields] AND (\"Chordata\"[Organism] OR chordata[All Fields])\" \\\n",
    "#    --o-sequences ./assign_taxo/ncbi-refseqs-unfiltered_Chordata.qza \\\n",
    "#    --o-taxonomy ./assign_taxo/ncbi-refseqs-taxonomy-unfiltered_Chordata.qza\n",
    "\n",
    "mkdir ./assign_taxo_Chordata\n",
    "\n",
    "qiime rescript get-ncbi-data \\\n",
    "    --m-accession-ids-file ./RefBase_Amph16S/IDs_uniq_NCBI_Mito_Chordata_1070F_1340R.txt \\\n",
    "    --o-sequences ./assign_taxo_Chordata/refseqs_NCBI_Mito_Chordata_1070F_1340R.qza \\\n",
    "    --o-taxonomy ./assign_taxo_Chordata/taxo_NCBI_Mito_Chordata_1070F_1340R.qza\n",
    "\n",
    "#qiime rescript get-ncbi-data \\\n",
    "#    --m-accession-ids-file ./RefBase_Amph16S/IDs_ENA_Amphibiens_BZH_Amph16S_1070F_1340R.txt \\\n",
    "#    --o-sequences ./assign_taxo_Amphibiens_BZH/refseqs_Amphibiens_BZH.qza \\\n",
    "#    --o-taxonomy ./assign_taxo_Amphibiens_BZH/taxo_Amphibiens_BZH.qza\n",
    "\n",
    "#qiime rescript get-ncbi-data \\\n",
    "#    --m-accession-ids-file ./RefBase_Amph16S/IDs_uniq_ENA_Amphibiens_invasifs_CROAA_1070F_1340R.txt \\\n",
    "#    --o-sequences ./assign_taxo_Amphibiens_tous/refseqs_Amphibiens_BZH_invasifs_CROAA_1070F_1340R.qza \\\n",
    "#    --o-taxonomy ./assign_taxo_Amphibiens_tous/taxo_Amphibiens_BZH_invasifs_CROAA_1070F_1340R.qza\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "a177cdba-9582-444a-8162-453b0e263db1",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@\n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@\n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@\n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@\n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@\n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@\n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@\n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@\n",
      "\u001b[?1l\u001b>\u001b[32mSaved FeatureData[Taxonomy] to: ./assign_taxo_Amphibiens_tous/blast_rep-seqs-97_Amph16S_Amphibiens_BZH_invasifs_CROAA.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureData[BLAST6] to: ./assign_taxo_Amphibiens_tous/results_rep-seqs-97_Amph16S_Amphibiens_BZH_invasifs_CROAA.qza\u001b[0m\n",
      "\u001b[0m\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@\n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@\n",
      "\u001b[?1l\u001b>\u001b[32mExported ./assign_taxo_Amphibiens_tous/blast_rep-seqs-97_Amph16S_Amphibiens_BZH_invasifs_CROAA.qza as TSVTaxonomyDirectoryFormat to directory ./assign_taxo_Amphibiens_tous/\u001b[0m\n",
      "\u001b[0m\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@\n",
      "\u001b[?1l\u001b>\u001b[?1h\u001b=@\n",
      "\u001b[?1l\u001b>\u001b[32mExported ./assign_taxo_Amphibiens_tous/results_rep-seqs-97_Amph16S_Amphibiens_BZH_invasifs_CROAA.qza as BLAST6DirectoryFormat to directory ./assign_taxo_Amphibiens_tous/\u001b[0m\n",
      "\u001b[0m\u001b[?1h\u001b=@(qiime2-amplicon-2024.5) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "cd /Users/tonyrobinet/sync/BioLav/M2_Ivan/Stage_M2/bioinfo\n",
    "\n",
    "# source activate qiime2-amplicon-2024.5\n",
    "\n",
    "qiime feature-classifier classify-consensus-blast \\\n",
    "        --i-query  rep-seqs-97_Amph16S.qza \\\n",
    "        --i-reference-reads ./assign_taxo_Amphibiens_tous/refseqs_Amphibiens_BZH_invasifs_CROAA_1070F_1340R.qza \\\n",
    "        --i-reference-taxonomy ./assign_taxo_Amphibiens_tous/taxo_Amphibiens_BZH_invasifs_CROAA_1070F_1340R.qza \\\n",
    "        --p-num-threads 8 \\\n",
    "        --p-perc-identity 0.7 \\\n",
    "        --p-query-cov 0.4 \\\n",
    "        --o-classification ./assign_taxo_Amphibiens_tous/blast_rep-seqs-97_Amph16S_Amphibiens_BZH_invasifs_CROAA.qza \\\n",
    "        --o-search-results ./assign_taxo_Amphibiens_tous/results_rep-seqs-97_Amph16S_Amphibiens_BZH_invasifs_CROAA.qza\n",
    "\n",
    "\n",
    "# exporter en .biom\n",
    "qiime tools export \\\n",
    "        --input-path ./assign_taxo_Amphibiens_tous/blast_rep-seqs-97_Amph16S_Amphibiens_BZH_invasifs_CROAA.qza \\\n",
    "        --output-path ./assign_taxo_Amphibiens_tous/\n",
    "\n",
    "qiime tools export \\\n",
    "        --input-path ./assign_taxo_Amphibiens_tous/results_rep-seqs-97_Amph16S_Amphibiens_BZH_invasifs_CROAA.qza \\\n",
    "        --output-path ./assign_taxo_Amphibiens_tous/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2b253beb-d610-480e-b240-962b5abcfb39",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Création d'une base de séquences références Eucaryotes sur le marqueur Amph16S\n",
    "# à partir de la base publique NCBI [mitochondrion AND 16S]\n",
    "\n",
    "cd /home/moustache/sync/BioLav/M2_Ivan/Stage_M2/bioinfo/RefBase_Amph16S/ncbi\n",
    "\n",
    "seqkit rmdup -s < ncbi_mitochondrion_AND_16S.fasta > ncbi_mitochondrion_AND_16S_nodup.fasta\n",
    "\n",
    "makeblastdb -in ncbi_mitochondrion_AND_16S_nodup.fasta -input_type fasta -dbtype nucl -parse_seqids -title ncbi_mitochondrion_AND_16S -hash_index -out ncbi_mitochondrion_AND_16S\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "a9047679-4c1d-43ab-b59f-25ef02d7d3ee",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n",
      "FASTA-Reader: Title ends with at least 20 valid nucleotide characters.  Was the sequence accidentally put in the title line?\n"
     ]
    }
   ],
   "source": [
    "cd /home/moustache/sync/BioLav/M2_Ivan/Stage_M2/bioinfo/\n",
    "\n",
    "blastn -db /home/moustache/sync/BioLav/M2_Ivan/Stage_M2/bioinfo/RefBase_Amph16S/ncbi/ncbi_mitochondrion_AND_16S \\\n",
    "    -query /home/moustache/sync/BioLav/M2_Ivan/Stage_M2/bioinfo/rep-seqs-97_Amph16S.fasta \\\n",
    "    -out ./assign_taxo_Eukaryota/rep-seqs-97_Amph16S_ncbi_mito.out"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
